crates/transpiler/src/passes/litinski_transformation.rs

```diff
@@ -88,119 +57,179 @@ pub fn run_litinski_transformation(
+
+                    let evo_qubits: Vec<Qubit> =
+                        indices.iter().map(|index| Qubit(*index)).collect();
+
+                    let label = z
+                        .iter()
+                        .rev()
+                        .zip(x.iter().rev())
+                        .map(|(z, x)| match (z, x) {
+                            (false, true) => 'X',
+                            (true, false) => 'Z',
+                            (true, true) => 'Y',
+                            (false, false) => {
+                                unreachable!("We do not produce 'I' paulis in the sparse format.");
+                            }
+                        })
+                        .collect::<String>();
+
+                    let py_pauli = PySparseObservable::from_label(label.as_str())?;
+
+                    let time = if sign {
+                        multiply_param(&angle, -0.5)
+                    } else {
+                        multiply_param(&angle, 0.5)
+                    };
+                    let py_evo = py_evo_cls.call1((py_pauli, time.clone()))?;
+                    let py_gate = PyGate {
+                        qubits: evo_qubits.len() as u32,
+                        clbits: 0,
+                        params: 1,
+                        op_name: "PauliEvolution".to_string(),
+                        gate: py_evo.into(),
+                    };
+
+                    new_dag.apply_operation_back(
+                        py_gate.into(),
+                        &evo_qubits,
+                        &no_clbits,
+                        Some(smallvec![time]),
+                        None,
+                        #[cfg(feature = "cache_pygates")]
+                        None,
+                    )?;
+                }
+                "measure" => {
+                    // Returns the evolved Pauli in the sparse format: (sign, pauli z, pauli x, indices),
+                    // where signs `true` and `false` correspond to coefficients `-1` and `+1` respectively.
+                    let (sign, z, x, indices) = clifford.get_inverse_z(qubits[0]);
+
+                    let ppm = PauliProductMeasurement { z, x, neg: sign };
+
+                    let ppm_qubits: Vec<Qubit> =
+                        indices.iter().map(|index| Qubit(*index)).collect();
+
+                    let ppm_clbits = dag.get_cargs(inst.clbits);
+
+                    new_dag.apply_operation_back(
+                        ppm.into(),
+                        &ppm_qubits,
+                        ppm_clbits,
+                        None,
+                        None,
+                        #[cfg(feature = "cache_pygates")]
+                        None,
+                    )?;
+                }
+                _ => unreachable!(
+                    "We cannot have unsupported names at this step of Litinski Transformation: {}",
+                    name
```

Comment:
Should this be a `panic!` given that it can be triggered by running on a circuit with unsupported gates? `unreachable!` should only be for paths that shouldn't be reachable from a logic perspective